### PR TITLE
Profile screen

### PR DESCRIPTION
**Pull request by manukyan-edgar**
[My profile screen](https://trello.com/c/66grn0tT/11-as-a-user-i-should-be-able-to-see-my-profile-screen)
[Change my password](https://trello.com/c/uJhZSMuH/20-as-a-user-i-should-be-able-to-change-my-password)

### CODE REVIEW

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/register/RegisterScreen.kt [hunk 1]
```diff
@@ -62,6 +63,8 @@ fun RegisterScreen(
     val registrationErrorState by registerViewModel.registrationErrorState.collectAsState()
     val registrationErrorMessage by registerViewModel.registerErrorMessage.collectAsState()
     val registerButtonEnabled by registerViewModel.enableRegisterButton.collectAsState()
+    val showLoading by registerViewModel.isOnLoadingState.collectAsState()
+    val inputMaxLength = 30
```
Comment (ManukyanQristine) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/register/RegisterScreen.kt]: inputMaxLength could be const

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt [hunk 2]
```diff
@@ -97,107 +100,114 @@ fun SignInScreen(
...
+                                    fontWeight = FontWeight(700),
+                                    color = AppTheme.colors.darkPink,
+                                )
+                            }
                         }
+                        AuthButton(
+                            modifier = Modifier
+                                .padding(horizontal = 32.dp, vertical = 16.dp)
+                                .fillMaxWidth(),
+                            shape = RoundedCornerShape(24.dp),
```
Comment (ManukyanQristine) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt]: You can use RoundedCorderShape defined on AppShapes.
for example AppTheme.shapes.mediumRoundedCorners for this case

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/forgotpassword/ForgotPasswordScreen.kt [hunk 3]
```diff
@@ -87,111 +83,117 @@ fun ForgotPasswordScreen(
...
                 }
                 AuthButton(
                     modifier = Modifier
                         .padding(horizontal = 32.dp, vertical = 24.dp)
-                        .padding(top = 100.dp)
                         .height(41.dp)
                         .width(200.dp),
-                    backgroundColor = AppTheme.colors.lightPink,
+                    backgroundColor = AppTheme.colors.white,
                     shape = RoundedCornerShape(13.dp),
```
Comment (ManukyanQristine) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/forgotpassword/ForgotPasswordScreen.kt:131]: You can use RoundedCorderShape defined on AppShapes.

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/register/RegisterScreen.kt (hunk 1 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/register/RegisterScreen.kt]: Fixed

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt (hunk 2 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt]: Fixed

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/forgotpassword/ForgotPasswordScreen.kt (hunk 3 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/forgotpassword/ForgotPasswordScreen.kt:131]: Fixed

File: app/src/main/java/com/simply/birthdayapp/data/mappers/UpdateProfileMapper.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.apollographql.apollo3.api.Optional
+import com.simply.birthdayapp.data.entities.UpdateProfileEntity
+import com.simply.birthdayapp.type.UpdateProfileInput
+
+fun UpdateProfileEntity.toUpdateProfile() =
+    UpdateProfileInput(
+        firstName = Optional.present(firstName),
+        lastName = Optional.present(lastName),
+        image = if (image != null) Optional.present(image) else Optional.absent(),
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/data/mappers/UpdateProfileMapper.kt]: You can use `Optional.presentIfNotNull()` here

File: app/src/main/java/com/simply/birthdayapp/di/DataModule.kt [hunk 5]
```diff
@@ -25,7 +27,7 @@ import org.koin.dsl.module
 
 val dataModule = module {
     single { DataStoreManager(context = androidContext()) }
-    single { createApolloClient(get()) }
+    factory { createApolloClient(get()) }
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/di/DataModule.kt:30]: Shouldn't ApolloClient be single?

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/components/BaseButton.kt [hunk 6]
```diff
@@ -28,20 +29,22 @@ fun AuthButton(
     enabled: Boolean = true,
     fontSize: TextUnit = 15.sp,
     textStyle: TextStyle = AppTheme.typography.bold,
+    textModifier:Modifier = Modifier,
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/components/BaseButton.kt]: Format the code. And it's a good practice to have just one modifier as an argument. For nested child views usually we don't pass modifiers from outside. In this case for text composable, which is inside the button.

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt [hunk 7]
```diff
@@ -97,107 +99,114 @@ fun SignInScreen(
...
-                                fontSize = 12.sp,
-                                fontFamily = FontFamily(Font(R.font.karm_light)),
-                                fontWeight = FontWeight(700),
-                                color = AppTheme.colors.darkPink,
-                            )
+                        Row(
+                            modifier = Modifier
+                                .height(32.dp)
+                                .fillMaxWidth(),
+                            Arrangement.End
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt]: Use named argument here

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ChangePasswordScreen.kt [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+    val hasRepeatPasswordError by profileViewModel.hasRepeatPasswordError.collectAsState()
+    val doneButtonEnabled by profileViewModel.enableDoneButton.collectAsState()
+    val changePasswordSuccess by profileViewModel.changePasswordSuccess.collectAsState()
+    val changePasswordErrorMessage by profileViewModel.changePasswordErrorMessage.collectAsState()
+    val changePasswordErrorState by profileViewModel.changePasswordErrorState.collectAsState()
+    val context = LocalContext.current
+
+    LaunchedEffect(changePasswordSuccess) {
+        if (changePasswordSuccess) {
+            Toast.makeText(context, (R.string.change_password_success), Toast.LENGTH_SHORT).show()
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ChangePasswordScreen.kt]: No need to have String ResID in brackets

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/EditAccountScreen.kt [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    val launcher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.GetContent()
+    ) { uri: Uri? ->
+        if (uri != null) profileViewModel.setImage(image = uri)
+    }
+
+    LaunchedEffect(updateProfileSuccess) {
+        if (updateProfileSuccess) {
+            Toast.makeText(context, (R.string.edit_account_success), Toast.LENGTH_SHORT).show()
+            daneClick()
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/EditAccountScreen.kt:73]: Fix typo here and the same fix with brackets

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/EditAccountScreen.kt [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+                    .fillMaxSize()
+                    .background(color = AppTheme.colors.backgroundPink)
+                    .verticalScroll(rememberScrollState()),
+                horizontalAlignment = Alignment.CenterHorizontally,
+            ) {
+                RoundAsyncImage(
+                    modifier = Modifier
+                        .padding(top = 50.dp)
+                        .height(150.dp)
+                        .width(150.dp)
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/EditAccountScreen.kt]: Use size here

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ProfileMainScreen.kt [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+            popUpTo(ProfileRootScreen.ProfileScreen.route) {
+                inclusive = true
+            }
+        }
+    }
+
+    fun navToEditAccountScreen() {
+        nestedNavController.navigate(ProfileRootScreen.EditAccountScreen.route) {
+            profileViewModel.setName(name = profileViewModel.profile.value?.firstName.toString())
+            profileViewModel.setSurName(surName = profileViewModel.profile.value?.lastName.toString())
```
Comment (Hayk985) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ProfileMainScreen.kt]: You can collect profile flow here to not always reference them from viewmodel

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt (hunk 7 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/SignInScreen.kt]: Fixed

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/EditAccountScreen.kt (hunk 10 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/EditAccountScreen.kt]: Fixed

File: app/src/main/java/com/simply/birthdayapp/data/mappers/UpdateProfileMapper.kt (hunk 4 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/data/mappers/UpdateProfileMapper.kt]: Fixed

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ChangePasswordScreen.kt (hunk 8 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ChangePasswordScreen.kt]: Fixed

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/components/BaseButton.kt (hunk 6 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/components/BaseButton.kt]: Fixed

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ProfileMainScreen.kt (hunk 11 above)
Comment (manukyan-edgar) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/main/profile/ProfileMainScreen.kt]: Fixed

**Review by Hayk985 — APPROVED** [review]

**Pull request merged**